File: Course11/deprecated_rn_backup/screens/HomeScreen.js
import * as React from 'react';
import { View, Text, Image, TextInput, Pressable, FlatList, StyleSheet, ActivityIndicator } from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';

const INITIAL_MENU_DATA = [
  {
    id: 1,
    name: 'Greek Salad',
    price: '$12.99',
    description: 'Our delicious salad is served with Feta cheese and peeled cucumber. Includes tomatoes, onions, olives, and extra virgin olive oil.',
    image: require('../assets/splash.png'),
    category: 'starters',
  },
  {
    id: 2,
    name: 'Bruschetta',
    price: '$7.99',
    description: 'Delicious grilled bread topped with garlic, tomatoes, olive oil and fresh basil.',
    image: require('../assets/splash.png'),
    category: 'starters',
  },
  {
    id: 3,
    name: 'Grilled Fish',
    price: '$20.00',
    description: 'Our tender grilled fish seasoned with fresh herbs and lemon zest, served with a side of roasted vegetables.',
    image: require('../assets/splash.png'),
    category: 'mains',
  },
  {
    id: 4,
    name: 'Pasta Carbonara',
    price: '$14.99',
    description: 'Classic Italian pasta with creamy egg sauce, pecorino cheese, crisp pancetta, and black pepper.',
    image: require('../assets/splash.png'),
    category: 'mains',
  },
  {
    id: 5,
    name: 'Lemon Dessert',
    price: '$6.99',
    description: 'Traditional homemade lemon tart with a buttery shortbread crust and delicate meringue topping.',
    image: require('../assets/splash.png'),
    category: 'desserts',
  },
  {
    id: 6,
    name: 'Baklava',
    price: '$5.99',
    description: 'Rich, sweet pastry made of layers of filo filled with chopped nuts and sweetened with honey syrup.',
    image: require('../assets/splash.png'),
    category: 'desserts',
  },
  {
    id: 7,
    name: 'House Wine',
    price: '$8.00',
    description: 'A glass of our signature house red or white wine sourced from Mediterranean vineyards.',
    image: require('../assets/splash.png'),
    category: 'drinks',
  },
  {
    id: 8,
    name: 'Fresh Lemonade',
    price: '$4.50',
    description: 'Refreshing squeezed lemonade made with mint leaves and a touch of organic honey.',
    image: require('../assets/splash.png'),
    category: 'drinks',
  },
];

const CATEGORIES = ['Starters', 'Mains', 'Desserts', 'Drinks'];

const HomeScreen = ({ navigation }) => {
  const [menuData, setMenuData] = React.useState(INITIAL_MENU_DATA);
  const [searchQuery, setSearchQuery] = React.useState('');
  const [selectedCategories, setSelectedCategories] = React.useState([]);
  const [userProfile, setUserProfile] = React.useState(null);

  React.useEffect(() => {
    const loadProfile = async () => {
      try {
        const firstName = await AsyncStorage.getItem('firstName');
        const lastName = await AsyncStorage.getItem('lastName');
        const avatar = await AsyncStorage.getItem('avatar');
        setUserProfile({
          firstName: firstName || 'User',
          lastName: lastName || '',
          avatar: avatar || null,
        });
      } catch (e) {
        console.error('Error loading profile in Home:', e);
      }
    };
    loadProfile();
    const unsubscribe = navigation.addListener('focus', loadProfile);
    return unsubscribe;
  }, [navigation]);

  const handleCategoryPress = (categoryName) => {
    const lower = categoryName.toLowerCase();
    if (selectedCategories.includes(lower)) {
      setSelectedCategories(selectedCategories.filter((c) => c !== lower));
    } else {
      setSelectedCategories([...selectedCategories, lower]);
    }
  };

  const filteredMenu = React.useMemo(() => {
    return menuData.filter((item) => {
      const matchesSearch =
        searchQuery.trim() === '' ||
        item.name.toLowerCase().includes(searchQuery.toLowerCase()) ||
        item.description.toLowerCase().includes(searchQuery.toLowerCase());
      const matchesCategory =
        selectedCategories.length === 0 ||
        selectedCategories.includes(item.category.toLowerCase());
      return matchesSearch && matchesCategory;
    });
  }, [menuData, searchQuery, selectedCategories]);

  const renderMenuItem = ({ item }) => (
    <View style={styles.menuItemContainer}>
      <View style={styles.menuItemDetails}>
        <Text style={styles.menuItemTitle}>{item.name}</Text>
        <Text style={styles.menuItemDescription} numberOfLines={2}>
          {item.description}
        </Text>
        <Text style={styles.menuItemPrice}>{item.price}</Text>
      </View>
      <Image source={item.image} style={styles.menuItemImage} resizeMode="cover" />
    </View>
  );

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <View style={styles.headerSpacer} />
        <Image
          source={require('../assets/little-lemon-logo.png')}
          style={styles.headerLogo}
          resizeMode="contain"
        />
        <Pressable style={styles.avatarButton} onPress={() => navigation.navigate('Profile')}>
          <Image
            source={require('../assets/profile.png')}
            style={styles.avatarImage}
          />
        </Pressable>
      </View>

      <View style={styles.heroSection}>
        <Text style={styles.heroTitle}>Little Lemon</Text>
        <View style={styles.heroContent}>
          <View style={styles.heroTextContainer}>
            <Text style={styles.heroSubtitle}>Chicago</Text>
            <Text style={styles.heroDescription}>
              We are a family owned Mediterranean restaurant, focused on traditional recipes served with a modern twist.
            </Text>
          </View>
          <Image
            source={require('../assets/splash.png')}
            style={styles.heroImage}
            resizeMode="cover"
          />
        </View>
        <View style={styles.searchContainer}>
          <TextInput
            style={styles.searchInput}
            placeholder="Search our delicious menu..."
            placeholderTextColor="#333333"
            value={searchQuery}
            onChangeText={setSearchQuery}
          />
        </View>
      </View>

      <View style={styles.breakdownSection}>
        <Text style={styles.breakdownTitle}>ORDER FOR DELIVERY!</Text>
        <View style={styles.categoriesContainer}>
          {CATEGORIES.map((category) => {
            const isSelected = selectedCategories.includes(category.toLowerCase());
            return (
              <Pressable
                key={category}
                style={isSelected ? styles.categoryButtonActive : styles.categoryButton}
                onPress={() => handleCategoryPress(category)}
              >
                <Text style={isSelected ? styles.categoryTextActive : styles.categoryText}>
                  {category}
                </Text>
              </Pressable>
            );
          })}
        </View>
      </View>

      <FlatList
        data={filteredMenu}
        keyExtractor={(item) => item.id.toString()}
        renderItem={renderMenuItem}
        contentContainerStyle={styles.listContent}
        ItemSeparatorComponent={() => <View style={styles.separator} />}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FFFFFF',
  },
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingHorizontal: 16,
    paddingVertical: 12,
    borderBottomWidth: 1,
    borderBottomColor: '#EE9972',
  },
  headerSpacer: {
    width: 44,
  },
  headerLogo: {
    width: 160,
    height: 40,
  },
  avatarButton: {
    width: 44,
    height: 44,
    borderRadius: 22,
    overflow: 'hidden',
    borderWidth: 1.5,
    borderColor: '#495E57',
    justifyContent: 'center',
    alignItems: 'center',
  },
  avatarImage: {
    width: 44,
    height: 44,
  },
  heroSection: {
    backgroundColor: '#495E57',
    paddingHorizontal: 16,
    paddingTop: 16,
    paddingBottom: 20,
  },
  heroTitle: {
    fontSize: 40,
    fontWeight: 'bold',
    color: '#F4CE14',
  },
  heroContent: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: 16,
  },
  heroTextContainer: {
    flex: 1,
    paddingRight: 16,
  },
  heroSubtitle: {
    fontSize: 24,
    fontWeight: '600',
    color: '#FFFFFF',
    marginBottom: 8,
  },
  heroDescription: {
    fontSize: 16,
    color: '#EDEFEE',
    lineHeight: 22,
  },
  heroImage: {
    width: 120,
    height: 120,
    borderRadius: 12,
  },
  searchContainer: {
    marginTop: 8,
  },
  searchInput: {
    backgroundColor: '#EDEFEE',
    height: 44,
    borderRadius: 8,
    paddingHorizontal: 16,
    fontSize: 16,
    color: '#333333',
  },
  breakdownSection: {
    paddingHorizontal: 16,
    paddingVertical: 16,
    borderBottomWidth: 1,
    borderBottomColor: '#EDEFEE',
  },
  breakdownTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    color: '#333333',
    marginBottom: 12,
  },
  categoriesContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  categoryButton: {
    backgroundColor: '#EDEFEE',
    paddingVertical: 8,
    paddingHorizontal: 12,
    borderRadius: 16,
  },
  categoryButtonActive: {
    backgroundColor: '#495E57',
    paddingVertical: 8,
    paddingHorizontal: 12,
    borderRadius: 16,
  },
  categoryText: {
    fontSize: 14,
    fontWeight: 'bold',
    color: '#495E57',
  },
  categoryTextActive: {
    fontSize: 14,
    fontWeight: 'bold',
    color: '#EDEFEE',
  },
  listContent: {
    paddingHorizontal: 16,
    paddingBottom: 24,
  },
  menuItemContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingVertical: 16,
  },
  menuItemDetails: {
    flex: 1,
    paddingRight: 16,
  },
  menuItemTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    color: '#333333',
    marginBottom: 4,
  },
  menuItemDescription: {
    fontSize: 14,
    color: '#495E57',
    marginBottom: 8,
    lineHeight: 20,
  },
  menuItemPrice: {
    fontSize: 16,
    fontWeight: '600',
    color: '#495E57',
  },
  menuItemImage: {
    width: 80,
    height: 80,
    borderRadius: 8,
  },
  separator: {
    height: 1,
    backgroundColor: '#EDEFEE',
  },
});

export default HomeScreen;

File: Course11/app/src/main/java/com/example/littlelemon/screens/HomeScreen.kt
package com.example.littlelemon.screens

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Search
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.navigation.NavHostController
import com.example.littlelemon.R

data class MenuItem(
    val id: Int,
    val name: String,
    val price: String,
    val description: String,
    val image: Int,
    val category: String
)

val INITIAL_MENU_DATA = listOf(
    MenuItem(1, "Greek Salad", "$12.99", "Our delicious salad is served with Feta cheese and peeled cucumber. Includes tomatoes, onions, olives, and extra virgin olive oil.", R.drawable.splash, "starters"),
    MenuItem(2, "Bruschetta", "$7.99", "Delicious grilled bread topped with garlic, tomatoes, olive oil and fresh basil.", R.drawable.splash, "starters"),
    MenuItem(3, "Grilled Fish", "$20.00", "Our tender grilled fish seasoned with fresh herbs and lemon zest, served with a side of roasted vegetables.", R.drawable.splash, "mains"),
    MenuItem(4, "Pasta Carbonara", "$14.99", "Classic Italian pasta with creamy egg sauce, pecorino cheese, crisp pancetta, and black pepper.", R.drawable.splash, "mains"),
    MenuItem(5, "Lemon Dessert", "$6.99", "Traditional homemade lemon tart with a buttery shortbread crust and delicate meringue topping.", R.drawable.splash, "desserts"),
    MenuItem(6, "Baklava", "$5.99", "Rich, sweet pastry made of layers of filo filled with chopped nuts and sweetened with honey syrup.", R.drawable.splash, "desserts"),
    MenuItem(7, "House Wine", "$8.00", "A glass of our signature house red or white wine sourced from Mediterranean vineyards.", R.drawable.splash, "drinks"),
    MenuItem(8, "Fresh Lemonade", "$4.50", "Refreshing squeezed lemonade made with mint leaves and a touch of organic honey.", R.drawable.splash, "drinks")
)

val CATEGORIES = listOf("Starters", "Mains", "Desserts", "Drinks")

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun HomeScreen(navController: NavHostController) {
    var searchQuery by remember { mutableStateOf("") }
    var selectedCategories by remember { mutableStateOf(emptySet<String>()) }

    fun handleCategoryPress(category: String) {
        val lower = category.lowercase()
        selectedCategories = if (selectedCategories.contains(lower)) {
            selectedCategories - lower
        } else {
            selectedCategories + lower
        }
    }

    val filteredMenu = INITIAL_MENU_DATA.filter { item ->
        val matchesSearch = searchQuery.isBlank() || item.name.contains(searchQuery, ignoreCase = true) || item.description.contains(searchQuery, ignoreCase = true)
        val matchesCategory = selectedCategories.isEmpty() || selectedCategories.contains(item.category.lowercase())
        matchesSearch && matchesCategory
    }

    Column(modifier = Modifier.fillMaxSize().background(Color.White)) {
        // Header
        Row(
            modifier = Modifier.fillMaxWidth().padding(horizontal = 16.dp, vertical = 12.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.SpaceBetween
        ) {
            Spacer(modifier = Modifier.width(44.dp))
            Image(
                painter = painterResource(id = R.drawable.little_lemon_logo),
                contentDescription = "Logo",
                modifier = Modifier.height(40.dp)
            )
            Image(
                painter = painterResource(id = R.drawable.profile),
                contentDescription = "Profile",
                modifier = Modifier
                    .size(44.dp)
                    .clip(CircleShape)
                    .clickable { navController.navigate("Profile") }
            )
        }

        // Hero Section
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .background(Color(0xFF495E57))
                .padding(16.dp)
        ) {
            Text(text = "Little Lemon", fontSize = 40.sp, fontWeight = FontWeight.Bold, color = Color(0xFFF4CE14))
            Row(modifier = Modifier.padding(bottom = 16.dp)) {
                Column(modifier = Modifier.weight(1f).padding(end = 16.dp)) {
                    Text(text = "Chicago", fontSize = 24.sp, fontWeight = FontWeight.SemiBold, color = Color.White)
                    Text(
                        text = "We are a family owned Mediterranean restaurant, focused on traditional recipes served with a modern twist.",
                        fontSize = 16.sp, color = Color(0xFFEDEFEE), lineHeight = 22.sp,
                        modifier = Modifier.padding(top = 8.dp)
                    )
                }
                Image(
                    painter = painterResource(id = R.drawable.splash),
                    contentDescription = "Hero Image",
                    contentScale = ContentScale.Crop,
                    modifier = Modifier.size(120.dp).clip(RoundedCornerShape(12.dp))
                )
            }
            OutlinedTextField(
                value = searchQuery,
                onValueChange = { searchQuery = it },
                placeholder = { Text("Search our delicious menu...") },
                leadingIcon = { Icon(Icons.Default.Search, contentDescription = "Search Icon") },
                modifier = Modifier.fillMaxWidth().background(Color(0xFFEDEFEE), RoundedCornerShape(8.dp)),
                singleLine = true,
                colors = OutlinedTextFieldDefaults.colors(
                    focusedBorderColor = Color.Transparent,
                    unfocusedBorderColor = Color.Transparent
                )
            )
        }

        // Breakdown Section
        Column(modifier = Modifier.padding(16.dp)) {
            Text(text = "ORDER FOR DELIVERY!", fontSize = 18.sp, fontWeight = FontWeight.Bold, color = Color(0xFF333333), modifier = Modifier.padding(bottom = 12.dp))
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween
            ) {
                CATEGORIES.forEach { category ->
                    val isSelected = selectedCategories.contains(category.lowercase())
                    Button(
                        onClick = { handleCategoryPress(category) },
                        colors = ButtonDefaults.buttonColors(
                            containerColor = if (isSelected) Color(0xFF495E57) else Color(0xFFEDEFEE),
                            contentColor = if (isSelected) Color(0xFFEDEFEE) else Color(0xFF495E57)
                        ),
                        shape = RoundedCornerShape(16.dp),
                        contentPadding = PaddingValues(horizontal = 12.dp, vertical = 8.dp)
                    ) {
                        Text(text = category, fontSize = 14.sp, fontWeight = FontWeight.Bold)
                    }
                }
            }
        }

        Divider(color = Color(0xFFEDEFEE), thickness = 1.dp)

        // Menu List
        LazyColumn(
            modifier = Modifier.fillMaxWidth(),
            contentPadding = PaddingValues(horizontal = 16.dp, vertical = 8.dp)
        ) {
            items(filteredMenu) { item ->
                Row(
                    modifier = Modifier.fillMaxWidth().padding(vertical = 16.dp),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Column(modifier = Modifier.weight(1f).padding(end = 16.dp)) {
                        Text(text = item.name, fontSize = 18.sp, fontWeight = FontWeight.Bold, color = Color(0xFF333333), modifier = Modifier.padding(bottom = 4.dp))
                        Text(text = item.description, fontSize = 14.sp, color = Color(0xFF495E57), modifier = Modifier.padding(bottom = 8.dp), maxLines = 2)
                        Text(text = item.price, fontSize = 16.sp, fontWeight = FontWeight.SemiBold, color = Color(0xFF495E57))
                    }
                    Image(
                        painter = painterResource(id = item.image),
                        contentDescription = item.name,
                        contentScale = ContentScale.Crop,
                        modifier = Modifier.size(80.dp).clip(RoundedCornerShape(8.dp))
                    )
                }
                Divider(color = Color(0xFFEDEFEE), thickness = 1.dp)
            }
        }
    }
}
